core/spec/models/spree/variant_spec.rb

```diff
@@ -773,6 +773,42 @@
     end
   end
 
+  describe '#shipping_category' do
+    context 'when shipping_category is nil' do
+      let(:product) { build(:product) }
+      let(:variant) { build(:variant, product: product, shipping_category_id: nil) }
+      it 'returns the parent products shipping_category' do
+        expect(variant.shipping_category).to eq(product.shipping_category)
+      end
+    end
+
+    context 'when shipping_category is set' do
+      let(:shipping_category) { create(:shipping_category) }
+      let(:variant) { build(:variant, shipping_category: shipping_category) }
+      it 'returns the shipping_category set on itself' do
+        expect(variant.shipping_category).to eq(shipping_category)
+      end
+    end
+  end
+
+  describe '#shipping_category_id' do
+    context 'when shipping_category_id is nil' do
+      let(:product) { build(:product) }
```

Comment:
Same as before. I'd explicitly add a shipping category to avoid false positives.

Reply:
done